File: file-translation-demo/nodejs-js/file-translate-result.js
const superagent = require('superagent');
const config = require('./config');

const authUtils = require('./utils/auth-utils');

const translateId = 'your translateId';
const url = `/api/v1/translate-result/${translateId}`;

const sendRequest = (serverConfig, accessKey, secretKey) => {
  const nonce = new Date().getTime().toString();
  const signature = authUtils.generateSignature(
    url,
    secretKey,
    nonce,
  );

  superagent.get(`${serverConfig.protocol}//${serverConfig.hostname}:${serverConfig.port}${url}`)
    .set({
      accessKey,
      signature,
      nonce,
    }).end((req, resp) => {
      console.log(resp.text);
    });
};

const main = async () => {
  try {
    await sendRequest(
      config.serverConfig,
      config.authConfig.accessKey,
      config.authConfig.secretKey,
    );
  } catch (error) {
    console.error(error);
  }
};

main();

File: file-translation-demo/python/file-translate-result.py
import time
import requests

import config
import auth_utils


def send_request(server_config, access_key, secret_key, translate_id):
    url = f"/api/v1/translate-result/{translate_id}"
    nonce = str(int(time.time()))
    signature = auth_utils.generate_signature(url, secret_key, nonce)

    headers = {
        "accessKey": access_key,
        "signature": signature,
        "nonce": nonce,
    }

    response = requests.get(
        f"{server_config['protocol']}//{server_config['hostname']}:{server_config['port']}{url}",
        headers=headers
    )
    print(response.text)

def main():
    try:
        send_request(
            config.server_config,
            config.auth_config['access_key'],
            config.auth_config['secret_key'],
            "your translateId", # response.json()['data']['translateId']
        )
    except Exception as error:
        print(error)

if __name__ == "__main__":
    main()
